runner/run.go

```diff
@@ -267,6 +268,14 @@ func needToSkipTest(include *regexp.Regexp, exclude *regexp.Regexp, testCase *sc
 	}
 
 	result := false
+	// if the test's tags do not match the passed ones
+	// it needs to be skipped
+	if tag != nil {
+		if !utils.MatchSlice(tag, testCase.Tags) {
+			result = true
```

Comment:
Just return from each of the condition. It doesn't make sense to evaluate further conditions once a condition has matched.

```suggestion
			return true
```